build-setup/linux.md

```diff
@@ -237,6 +237,68 @@ sudo make install
 
 to complete the build.
 
+### CMake
+
+CMake version 3.5 is the minimum required to build PyTorch. Download version 3.19.3 from <https://github.com/Kitware/CMake/releases/download/v3.19.3/cmake-3.19.3-Linux-x86_64.sh> and install:
+
+```
+chmod +x cmake-3.19.3-Linux-x86_64.sh
+sudo mkdir /usr/local/cmake
+sudo ./cmake-3.19.3-Linux-x86_64.sh --skip-license --prefix=/usr/local/cmake
+```  
+
+In your `.bashrc` file add CMake to `PATH`:
+
+```
+PATH=$PATH:/usr/local/cmake/bin
+```
+
+### Python 
+
+PyTorch currently requires Python 3.6, 3.7 or 3.8, and version 3.7 appears to cause fewest problems in their test status matrix, so we use that. If your system does not have a requisite version of Python install it with a package manager or build the last 3.7 release from source by downloading `Python-3.7.9.tgz` from <https://www.python.org/ftp/python/3.7.9/Python-3.7.9.tgz> then extract and build:
+
+```
+tar -xzf Python-3.7.9.tgz
+cd Python-3.7.9/
+./configure --enable-optimizations
+make
+sudo make altinstall
```

Comment:
Reading the make file the `altinstall` target does not overwrite any current installations by updating softlinks etc. We will use altinstall and be explicit when invoking python